### PR TITLE
*: clear Transport trait, and split with another RaftHandler trait.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @disksing

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL

**Comment by ngaut** [comment]
LGTM

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -89,8 +89,8 @@ fn get_store_path(matches: &Matches) -> String {
 
 fn run_local_server(listener: TcpListener, store: Storage) {
     let mut event_loop = create_event_loop().unwrap();
-    let trans = Arc::new(RwLock::new(MockTransport));
-    let mut svr = Server::new(&mut event_loop, listener, store, trans).unwrap();
+    let raft_router = Arc::new(RwLock::new(MockRaftStoreRouter));
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: `router` or `store_router` may be better.

File: src/server/server.rs [hunk 2]
```diff
@@ -49,12 +50,12 @@ pub struct Server<T: Transport> {
     // addr -> Token
     peers: HashMap<String, Token>,
 
-    trans: Arc<RwLock<T>>,
+    raft: Arc<RwLock<T>>,
```
Comment (BusyJay) [inline src/server/server.rs]: `router`?

File: src/bin/tikv-server.rs (hunk 1 above)
Comment (siddontang) [inline src/bin/tikv-server.rs]: we use raft to emphasize this is a router for raft store. So I think it is ok. 

@ngaut

File: src/storage/engine/raftkv.rs [hunk 3]
```diff
@@ -81,10 +82,10 @@ impl<T: PdClient, Trans: Transport> RaftKv<T, Trans> {
         let (tx, rx) = mpsc::channel();
         let uuid = req.get_header().get_uuid().to_vec();
         let l = req.get_requests().len();
-        try!(self.trans.rl().send_command(req,
-                                          Box::new(move |r| {
-                                              tx.send(r).map_err(::raftstore::errors::other)
-                                          })));
+        try!(self.raft.rl().send_command(req,
+                                         Box::new(move |r| {
```
Comment (BusyJay) [inline src/storage/engine/raftkv.rs]: `box` is enough.

File: src/bin/tikv-server.rs (hunk 1 above)
Comment (ngaut) [inline src/bin/tikv-server.rs]: router is fine to me.

File: src/server/server.rs (hunk 2 above)
Comment (siddontang) [inline src/server/server.rs]: Router may be too short, so I prefer raft_router here.

**Comment by BusyJay** [comment]
rest LGTM

**Comment by BusyJay** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**